teps/0015-implicit-mapping.md

```diff
@@ -0,0 +1,658 @@
+|2| Tekton tries to resolve omitted parameters and resources for a pipeline `Task` in the following order: <br><br> <ol><li> If any preceding `Task` has a `Task` result whose name matches an implicit parameter name of the `Task`, resolve the implicit parameter value with the last produced `Task` result that matches. <li> If a parameter or resource name in the `Pipeline.spec` field matches a `Task's` implicit parameter or resource name, resolve the parameter and/or resource values. </li> <li> Tekton produces an error if parameters or resources can't be matched. <br> *\*Explicitly specified params and resources always take precedence.*
+
+[DISCUSSION]: Additionally, omitting input resources that use the output resource of a previous task by using the [`from`](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-the-from-parameter) parameter could be supported.
+This could mean making the `from` parameter optional.
+When omitted, Tekton can match an input resource to the latest matching output resource.
+
+## Proposal
+
+Allow for an implicit specification of a `Pipeline` wherein parameters and resources for a `PipelineTask` can be omitted.
+
+<!-- #### User Story
+As a `Pipeline` author, writing a `Pipeline` specification with many `Task`s that have parameters and resources can become tedious.
+The `Pipeline` specification can become verbose and contains redundant information.
+With implicit parameter and resource mapping I can omit parameters and resources for PipelineTasks.
+As a result, writing a `Pipeline` specification becomes less tedious, and the `Pipeline` specification is easier to read due to reduced verbosity.
+--->
+
+### Notes
+
+- The VSCode Tekton plugin need not be updated to support implicit `Pipeline` specifications
+- Potentially, the Tekton Dashboard needs to be updated to support implicit `Pipeline` specifications
+
+### Caveats
+#### Common parameter and resource names
+Certain parameter or resource names may be relatively common and therefore carry a different meaning between different `Tasks`.
+`Tasks` with the same parameter and resource names that require different input could therefore inadvertently acquire the wrong value.
+We consider it the `Pipeline` author's responsibility to guard against these errors. 
+
+#### Non-standardized parameter and resource names
+Since parameter and resource names are not standardized, it may occur that a `Pipeline` parameter or resource should be passed to differently named parameters and resources across `Tasks`.
+To combat this, we could propose an alias for `Pipeline` parameters and resources in order to match parameters and resources across multiple `Tasks`.
```

Comment:
If we add this, it would make sense to add it in the Pipeline's `spec.params` field and thus altering the `ParamSpec`. Modifying the `ParamSpec` would mean that both `Pipeline` and `Task` authors can specify aliases for parameters. 

To restrict this feature to `Pipeline` authors, who'd have to know the different parameter names, we would have to separate the `ParamSpec` type for `Pipeline` and `Task` types.
Alternatively,  we could avoid separating the `ParamSpec` if the webhook rejects `Tasks` with alias fields.